lib/src/signed_video_h26x_auth.c

```diff
@@ -201,7 +201,7 @@ prepare_for_link_and_gop_hash_verification(signed_video_t *self, h26x_nalu_list_
     item = item->next;
   }
 
-  if (list_idx + hash_size > self->gop_info->hash_list_size) {
+  if (list_idx + hash_size > self->gop_info->hash_list_size + 1) {
```

Comment:
This will not solve the bug since you will still enter this if-statement with the example I provided. Have you tested it?